### PR TITLE
Feat: add modals

### PR DESCRIPTION
**Pull request by gustavosf2003**
This PR adds modals by pressing the product card. Note that the description content has not yet been created in cms

image

### CODE REVIEW

File: src/components/CustomModal.tsx [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  children,
+}: CustomModalProps) => {
+  const toggleModal = () => {
+    setIsOpen(!isOpen);
+  };
+  return (
+    <Modal onBackdropPress={toggleModal} isVisible={isOpen}>
+      <View className="rounded-md bg-slate-200 h-2/3">
+        <View className="flex flex-row items-center justify-between px-4 pt-4 pb-3 bg-white border-b border-gray-400 rounded-md">
+          <Text className="text-xl ">{title}</Text>
```
Comment (kkfuri) [inline src/components/CustomModal.tsx]: space after class

File: src/components/ProductModal.tsx [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+  counterValue,
+  setCounterValue,
+}: ProductModalProps) => {
+  const { savedProducts, setSavedProducts } = useContext(ProductContext);
+  const productContent = product.content;
+  return (
+    <CustomModal isOpen={isOpen} setIsOpen={setIsOpen} title={`#${product.id}`}>
+      <>
+        <ScrollView>
+          <View className="mx-2 mt-6 ">
```
Comment (kkfuri) [inline src/components/ProductModal.tsx]: space after class

File: src/components/ProductModal.tsx [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+  product,
+  isOpen,
+  setIsOpen,
+  counterValue,
+  setCounterValue,
+}: ProductModalProps) => {
+  const { savedProducts, setSavedProducts } = useContext(ProductContext);
+  const productContent = product.content;
+  return (
+    <CustomModal isOpen={isOpen} setIsOpen={setIsOpen} title={`#${product.id}`}>
```
Comment (kkfuri) [inline src/components/ProductModal.tsx]: No one cares about product id when purchasing, I think the title could be the name and having the id as a small SKU info on the product descript

File: src/components/ProductModal.tsx [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+                <Text className="font-bold">
+                  {productContent.price}
+                  {businessRules.currency}
+                </Text>
+              </View>
+            </View>
+            <View className="mt-3">
+              <Text className="text-lg font-medium">Description</Text>
+              <Text className="mt-3">
+                Lorem ipsum dolor, sit amet consectetur adipisicing elit.
```
Comment (kkfuri) [inline src/components/ProductModal.tsx]: Why use hard coded mock when you can mock the prop

**Review by kkfuri — CHANGES_REQUESTED** [review]

File: src/components/ProductModal.tsx (hunk 3 above)
Comment (Pedrodev3) [reply src/components/ProductModal.tsx]: I totally agree

**Review by vitor-afonso — APPROVED** [review]

**Comment by Pedrodev3** [comment]
Maybe if you change the title and subtitle to left and the picture to right using space-between, and after that use a minimal padding-x I believe the layout would improve and fit better in container

File: src/components/ProductModal.tsx (hunk 3 above)
Comment (gustavosf2003) [reply src/components/ProductModal.tsx]: Make sense 🫡

File: src/components/ProductModal.tsx (hunk 4 above)
Comment (gustavosf2003) [reply src/components/ProductModal.tsx]: Changed to real value

**Comment by gustavosf2003** [comment]
> Maybe if you change the title and subtitle to left and the picture to right using space-between, and after that use a minimal padding-x I believe the layout would improve and fit better in container

I did it another way, the last design was terrible so I did some research and implemented it like this:
image

**Comment by vitor-afonso** [comment]
I like this one ❤️

File: src/components/ProductCard.tsx [hunk 5]
```diff
@@ -30,56 +32,69 @@ const ProductCard = ({ product, quantity, showDynamicPrice }: ProductProp) => {
...
+          <View className="flex justify-between my-1 ml-2 ">
+            <View>
+              <Text className="flex font-medium">{productContent.name}</Text>
+              <Text className="text-gray-700">{productContent.unit}</Text>
+              <Text className="text-gray-700">
+                Origin: {productContent.origin}
+              </Text>
+            </View>
+            <Text className="font-bold">
+              {parseCurrency(
```
Comment (vitor-afonso) [inline src/components/ProductCard.tsx:73]: Wouldn't be better to have this logic in a function or variable and then just use it here to make readability easier?

File: src/components/ProductCard.tsx [hunk 6]
```diff
@@ -30,56 +32,69 @@ const ProductCard = ({ product, quantity, showDynamicPrice }: ProductProp) => {
...
+                showDynamicPrice
+                  ? productContent.price * counterValue
+                  : productContent.price
+              )}
```
Comment (vitor-afonso) [inline src/components/ProductCard.tsx:77]: Wouldn't be better to have this logic in a function or variable and then just use it here to make readability easier?

**Pull request merged**